### PR TITLE
Speaker is also able to see the proposal and able to modify it

### LINKED ISSUE #1482: Show proposals to all speakers, not just submitter
Users are able to see their submissions, but that only works if the user is the submitter. It should be showing the proposals where the user is just the speaker as well.

### PR DESCRIPTION
**Pull request by namangupta01**
Closes #1482

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -345,24 +345,35 @@
     end
 
     describe 'proposals methods' do
-      let(:submitter) { create(:submitter, user: user) }
```
Comment (Ana06) [inline spec/models/user_spec.rb]: How was that working before? where is `submitter` defined? 🤔

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -345,24 +345,35 @@
...
+      let(:submitter) { create(:user) }
+      let(:speaker) { create(:user) }
       let(:event1) { create(:event, program: conference.program) }
       let(:event2) { create(:event, program: conference.program) }
+      let(:event3) { create(:event, program: conference.program) }
```
Comment (Ana06) [inline spec/models/user_spec.rb]: Why do you need a third event for that, use one of the others. The case when we have both a submitter and an speaker is the one we want to test. 😉

File: spec/models/user_spec.rb (hunk 1 above)
Comment (namangupta01) [reply spec/models/user_spec.rb]: It is defined in factory file 'event_users.rb'

File: spec/models/user_spec.rb (hunk 2 above)
Comment (namangupta01) [reply spec/models/user_spec.rb]: Yes you are right i will correct it!

**Comment by namangupta01** [comment]
@Ana06 respective changes are done. Please have a look!

**Comment by differentreality** [comment]
I think this looks good. @Ana06 ?

**Comment by bear454** [comment]
Please update your review @Ana06 or it will be dismissed.

**Comment by differentreality** [comment]
Thank you @namangupta01 :+1: 
Sorry this took so long!

**Pull request merged**